actionpack/lib/action_dispatch/http/mime_type.rb

```diff
@@ -136,6 +136,7 @@ def self.find_item_by_name(array, name)
     class << self
       TRAILING_STAR_REGEXP = /^(text|application)\/\*/
       PARAMETER_SEPARATOR_REGEXP = /;\s*\w+="?\w+"?/
+      HEADER_SEPARATOR_REGEXP = /(?!".*),(?!.*")/
```

Comment:
There are a lot of different ways to identify a comma _not_ within quotes. I opted for the most simple approach, but I'm happy to explore other routes and measure their speed differences.